### PR TITLE
feat: Increase maximum concurrent requests for the server to 1000

### PR DESCRIPTION
**Pull request by kappratiksha**
Increase maximum concurrent requests for the server to 1000 and some buffer

### CODE REVIEW

File: lib/functions_framework/server.rb [hunk 1]
```diff
@@ -306,7 +306,7 @@ def min_threads
       # @return [Integer]
       #
       def max_threads
-        @max_threads || 8
+        @max_threads || 1024
```
Comment (jihuin) [inline lib/functions_framework/server.rb:309]: Do you think it's necessary to also add some buffer to @max_threads here?

Comment (anniefu) [reply lib/functions_framework/server.rb:309]: Probably, Puma takes some threads by default: https://github.com/puma/puma#thread-pool

Comment (kappratiksha) [reply lib/functions_framework/server.rb:309]: I did run the conformance concurrency test with 1000 requests and it passed with 1024 threads so decided to go ahead with this number.

_"Be aware that additionally Puma creates threads on its own for internal purposes (e.g. handling slow clients). So, even if you specify -t 1:1, expect around 7 threads created in your application."_ - Looks like Puma adds extra threads beyond the specified limit if needed.

Comment (kappratiksha) [reply lib/functions_framework/server.rb:309]: Tested this code path by setting the --max-thread=x parameter and setting the x to 10, 20, 50, 100, 200.. and the tests worked without needing any buffer.

**Comment by anniefu** [comment]
Uh-oh it looks like the newest GitHub runner for `windows-latest` was changed 2 days ago and doesn't seem to have `bundle` installed anymore... https://github.com/actions/runner-images/commit/4994130a965d5529b240818de583d48febefcfdf

**Review by anniefu — APPROVED** [review]

**Pull request merged**